src/Http/Session.php

```diff
@@ -491,8 +502,18 @@ public function consume(string $name)
      */
     public function write($name, $value = null): void
     {
-        if (!$this->started()) {
-            $this->start();
+        $started = $this->started() || $this->start();
+        if (!$started) {
+            $message = 'Could not start the session';
+            if ($this->headerSentInfo !== null) {
+                $message .= sprintf(
+                    ', headers already sent in file `%s` on line `%s`',
+                    $this->headerSentInfo['filename'],
```

Comment:
```suggestion
                    Debugger::trimPath($this->headerSentInfo['filename']),
```
\+ `use Cake\Error\Debugger;`